src/nvim/window.c

```diff
@@ -3750,6 +3758,8 @@ int win_new_tabpage(int after, char_u *filename)
 
     tabpage_check_windows(tp);
 
+    lastused_tabpage = tp;
+
     apply_autocmds(EVENT_WINNEW, NULL, NULL, false, curbuf);
     apply_autocmds(EVENT_WINENTER, NULL, NULL, false, curbuf);
     apply_autocmds(EVENT_TABNEW, filename, filename, false, curbuf);
```

Comment:
I wonder if `EVENT_WINNEW` should fire _before_ the last-used tabpage is updated. Let's just go with this for now...